<launch>
  
  <arg name="uav_name" default="$(optenv UAV_NAME uav91)"/>
  <arg name="visualize" default="false"/>

  <group ns="$(arg uav_name)" >

    <node pkg="label_extraction" type="extract_labels.py" name="label_extractor" output="screen" >
      
      <param name="mask_filename" value="$(find label_extraction)/masks/eaglemk2_os0_mask.bmp" />

      <!-- input topics -->
      <remap from="lidar_tracker/tracks"             to="lidar_tracker/tracks" />
      <remap from="os_img_nodelet/ambient_image"     to="os_img_nodelet/ambient_image" />
      <remap from="os_img_nodelet/intensity_image"   to="os_img_nodelet/intensity_image" />
      <remap from="os_img_nodelet/range_image"       to="os_img_nodelet/range_image" />

      <!-- output topics -->
      <remap from="labels" to="/dataset/labels" />

    </node>

    <node if="$(arg visualize)" pkg="label_extraction" type="visualize_labels.py" name="label_visualizer" output="screen" >
      
      <!-- input topics -->
      <remap from="labels_in" to="/dataset/labels" />
      <remap from="os_img_nodelet/ambient_image"     to="os_img_nodelet/ambient_image" />
      <remap from="os_img_nodelet/intensity_image"   to="os_img_nodelet/intensity_image" />
      <remap from="os_img_nodelet/range_image"       to="os_img_nodelet/range_image" />

      <!-- output topics -->
      <remap from="labels_vis" to="/dataset/labels_vis" />

    </node>

  </group>

</launch>
